### ISSUE TITLE
Can't launch FanControl on Windows 11 due to .NET Core Runtime v8 missing, even though it's installed?

### ISSUE THREAD

**Issue by pcgeek86** [Rem0o/FanControl.Releases#2864]
**Describe the bug**

**UPDATE**: The .NET 4.8 version runs okay. Not sure why the .NET Core version won't run though.

I tried running FanControl on my Windows 11 (production release, not preview / beta) system, and received the message about .NET Core Runtime v8 below. I already have this installed on my system, so I chose to "repair" it, which succeeded, but FanControl still won't launch. Any ideas on how to get FanControl to launch?

I tried installing FanControl from the Scoop package manager first, and then uninstalled it from Scoop, and tried downloading the ZIP release from GitHub directly, and running that. The result is the same regardless of how I attempt to install it.

![image](https://github.com/user-attachments/assets/98a7653d-09e5-43b5-86bd-15e4c2fe7540)

![image](https://github.com/user-attachments/assets/d0fafae8-e29f-433e-be6c-bd041fccc0fd)

```
[Window Title]
FanControl.exe

[Main Instruction]
You must install or update .NET to run this application.

[Expanded Information]
Required: 'Microsoft.NETCore.App', version '8.0.0' (x64)

Learn more:
https://aka.ms/dotnet/app-launch-failed

Download link:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&&framework_version=8.0.0&&arch=x64&&rid=win-x64&&os=win10&&gui=true

[^] Hide details  [Download it now] [Close]
```

**Is there a log.txt file next to FanControl.exe with recent date entries?**
[Link it here]

There is no log file generated. The application does not start up.

**Relevant hardware specs and setup**

- **Motherboard**: ASRock X570 AM4 Phantom Gaming 4 WiFi AX
- **CPU**: AMD Ryzen 9 3950X (16 cores, 32 threads)
- **Memory**: Corsair Vengeance RGB 64GB DDR4
- **Disk**: WD Blue SN5000 4 TB
- **GPU**: NVIDIA GeForce RTX 4070 Ti Super 16 GB (Gigabyte Windforce)

**Comment by Rem0o** [comment]
Have you tried downloading from the link in the dialog and installing through ?
Pretty sure your existing .NET install has some kind of problem with it, be it because it's installed to a specific user, isn't installed in  the expected location, an environment variable is wrong, or I don't know, but there isn't anything wrong related to the .NET 8 build itself.

**Comment by pcgeek86** [comment]
Yup I downloaded the .NET runtime from the link, and ran it. That told me I already had it installed, so I went ahead and repaired it, and it said everything was fine.

**Comment by Rem0o** [comment]
Try and uninstall it completely ?

**Comment by pcgeek86** [comment]
> Try and uninstall it completely ?

Doing that now

**Comment by Rem0o** [comment]
Never had this kind of issue, so I can't say without doing some research on the subject. In any case I don't think it's a "FanControl" problem, but more of a dotnet problem in your case, would bet any .NET 8 app would do the same thing in your actual environment.

**Comment by pcgeek86** [comment]
Agreed - I'll explore the .NET docs and see what I can find out. Thanks for the ideas.

**Comment by pcgeek86** [comment]
I figured this out today finally. Had a similar problem with another application, called Lively Wallpaper.

It turns out that the `DOTNET_ROOT` environment variable was set to a custom path, because I had the `dotnet-sdk` package installed using the Scoop package manager. https://scoop.sh

Rather than overriding the environment variable, I simply removed that package from my system, which unset the variable and allowed the Windows / .NET defaults to apply.

After doing this, the installed .NET Runtimes were discoverable in their default paths under `Program Files`.

https://learn.microsoft.com/en-us/dotnet/core/runtime-discovery/troubleshoot-app-launch
